description.md

```diff
@@ -0,0 +1,937 @@
+# 32  bytes of permit
+# 32  bytes of custom data
+```
+
+### **Non-ERC20 tokens swap**
+
+To swap tokens with a transfer function signature that do not comply with the IERC20 interface `IERC20.transferFrom(from, to, amount)`, you need to use a proxy contract. The proxy contract takes `from`, `to`, and `amount` arguments, maps them to the token transfer function, and adds additional parameters from the suffix calldata.
+
+In this case, `makerAsset` and/or `takerAsset` should contain the address of the proxy contract, and the token address should be encoded into the suffix with additional arguments. The suffix consists of packed bytes representing the calldata of parameters that are passed to the proxy contract.
+
+Examples of tokens that have already implemented this method are based on the following standards:
+
+- ERC721 - `IERC721.transferFrom(from, to, tokenId)`
+- ERC1155 - `IERC1155.safeTransferFrom(from, to, tokenId, amount, data)`
+
+To implement your custom proxy:
+
+1. Implement a function that receives `from`, `to`, and `amount` as the first three parameters, along with additional parameters that will be passed with `makerAssetSuffix` and/or `takerAssetSuffix`.
+2. Find the function name whose selector equals `transferFrom(from,to,amount)`.
+3. Implement `transferFrom` for the token.
+
+**Implementation example**
+
+In this example, we are using an ERC721 token proxy contract:
+
+```solidity
+contract ERC721Proxy is ImmutableOwner {
+    error ERC721ProxyBadSelector();
+
+    constructor(address _immutableOwner) ImmutableOwner(_immutableOwner) {
+        if (ERC721Proxy.func_60iHVgK.selector != IERC20.transferFrom.selector) revert ERC721ProxyBadSelector();
+    }
+
+    /// @notice Proxy transfer method for `IERC721.transferFrom`. Selector must match `IERC20.transferFrom`.
+    /// Note that `amount` is unused for security reasons to prevent unintended ERC-721 token sale via partial fill
+    // keccak256("func_60iHVgK(address,address,uint256,uint256,address)") == 0x23b872dd (IERC20.transferFrom)
+    function func_60iHVgK(address from, address to, uint256 /* amount */, uint256 tokenId, IERC721 token) external onlyImmutableOwner {
+        token.transferFrom(from, to, tokenId);
+    }
+}
+```
+
+As you can see, the `amount` parameter is not used, but the function still needs to accept it. Additionally, two extra parameters, `tokenId` and `token`, are passed to the function to perform the `transferFrom` operation.
+
+**To use this proxy, the order should:**
+
+1. Include the proxy address in the `makerAsset` or `takerAddress` field.
+2. Have the `HAS_EXTENSION` flag set.
+3. Include an extension with either `makerAssetSuffix` or `takerAssetSuffix` set, with the packed `tokenId` and `token` address. This information will be passed to the proxy when the order is filled.
+
+> Note: This extension is incompatible with the `USE_PERMIT2_FLAG` setting.
+> 
+
+**Example**
+
+Here is the code that creates an order:
+
+```javascript
```

Comment:
Maybe we should rename `addr1` in example and explain where users can get `buildOrder` method

Reply:
These examples are located in tests/examples/LimitOrderProtocol-example.js. If a reader needs it, he can find the implementation in code.